_config/email.yml

```diff
@@ -1,12 +1,55 @@
 ---
-Name: emailconfig
+Name: mailer
```

Comment:
I just changed it to be consistent with all of the `.mailer` suffixed I added to the service key names below

In the changelog we'll need to explicitly call out that the email config is completely different, not the the `after` bit.  The new `after` bit is now `mailer-dsn-default-config` anyway

Reply:
I don't think we need to rename this just because the below config is suffixed as `.mailer` though. Aside from the fact that I don't understand why they're suffixed in the first place which will be addressed in a separate question below, I don't think this coupling is useful.

Reply:
It should be `mailer` for consistency with everything else.  The main reason I called it `mailer` is to sync it up the fact this all relates to symfony/**mailer**.  I've since renamed email.yml to mailer.yml for consistency too.

Even if we retained this config as `emailconfig`, to facilitate projects keeping the same `After:` in a project while updading, it still wouldn't work because the bit in needs to be `After` is `mailer-dsn-default-config`, not `mailer`

If this was the goal, we'd need to rename `mailer-dsn-default-config` to `emailconfig`, which just makes the rest of the yml config confusing.

I proper solution here is clearly tell people in the upgrade guide: "Email needs to be completely reconfigured, read the instructions"

Reply:
> Even if we retained this config as emailconfig, to facilitate projects keeping the same `After:` in a project while updading, it still wouldn't work because the bit in needs to be After is `mailer-dsn-default-config`, not `mailer`

In that case I'd recommend changing `mailer-dsn-default-config` to `emailconfig` but I can see your point and ultimately I'm not going to keep bouncing this back and forth just for this. I'm okay with leaving the naming as you have it currently.

I've added a note to the changelog to add a very clear note to say the config has changed completely as per your suggestion ('The proper solution here is clearly tell people in the upgrade guide: "Email needs to be completely reconfigured, follow the very clear instructions we are providing"')